x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/SpatialIntersects.java

```diff
@@ -0,0 +1,288 @@
+        evaluatorMap.put(
+            key.withConstants(false, true),
+            new SpatialEvaluatorWithConstantFactory(SpatialIntersectsCartesianStringAndConstantEvaluator.Factory::new)
+        );
+        evaluatorMap.put(
+            key.withConstants(true, true),
+            new SpatialEvaluatorWithConstantsFactory(SpatialIntersectsCartesianConstantAndConstantEvaluator.Factory::new)
+        );
+        return evaluatorMap;
+    }
+
+    @Evaluator(extraName = "GeoSourceAndConstant", warnExceptions = { IllegalArgumentException.class, IOException.class })
+    static boolean processGeoSourceAndConstant(BytesRef leftValue, @Fixed Component2D rightValue) throws IOException {
+        return GEO.geometryRelatesGeometry(leftValue, rightValue);
+    }
+
+    @Evaluator(extraName = "GeoSourceAndString", warnExceptions = { IllegalArgumentException.class, IOException.class })
+    static boolean processGeoSourceAndString(BytesRef leftValue, BytesRef rightValue) throws IOException {
+        BytesRef geometry = SpatialCoordinateTypes.UNSPECIFIED.wktToWkb(rightValue.utf8ToString());
+        return GEO.geometryRelatesGeometry(leftValue, geometry);
+    }
+
+    @Evaluator(extraName = "GeoSourceAndSource", warnExceptions = { IllegalArgumentException.class, IOException.class })
+    static boolean processGeoSourceAndSource(BytesRef leftValue, BytesRef rightValue) throws IOException {
+        return GEO.geometryRelatesGeometry(leftValue, rightValue);
+    }
+
+    @Evaluator(extraName = "GeoPointDocValuesAndConstant", warnExceptions = { IllegalArgumentException.class })
+    static boolean processGeoPointDocValuesAndConstant(long leftValue, @Fixed Component2D rightValue) {
+        return GEO.pointRelatesGeometry(leftValue, rightValue);
+    }
+
+    @Evaluator(extraName = "GeoPointDocValuesAndString", warnExceptions = { IllegalArgumentException.class })
+    static boolean processGeoPointDocValuesAndString(long leftValue, BytesRef rightValue) {
+        Geometry geometry = SpatialCoordinateTypes.UNSPECIFIED.wktToGeometry(rightValue.utf8ToString());
+        return GEO.pointRelatesGeometry(leftValue, geometry);
+    }
+
+    @Evaluator(extraName = "GeoPointDocValuesAndSource", warnExceptions = { IllegalArgumentException.class })
+    static boolean processGeoPointDocValuesAndSource(long leftValue, BytesRef rightValue) {
+        Geometry geometry = SpatialCoordinateTypes.UNSPECIFIED.wkbToGeometry(rightValue);
+        return GEO.pointRelatesGeometry(leftValue, geometry);
+    }
+
+    @Evaluator(extraName = "GeoConstantAndConstant", warnExceptions = { IllegalArgumentException.class, IOException.class })
+    static boolean processGeoConstantAndConstant(@Fixed GeometryDocValueReader leftValue, @Fixed Component2D rightValue)
+        throws IOException {
+        return GEO.geometryRelatesGeometry(leftValue, rightValue);
```

Comment:
I'm pretty sure this shouldn't exist. if all the arguments are `@Fixed` then constant folding should kick in and we'll use one of the other evaluators to make a boolean one time.

Reply:
Right, so if both sides are constant, we don't need the generated code, or reading blocks, at all. I know my integration tests hit this code path, so I need to figure out how to make the shortcut happen. I'll investigate.

Reply:
there's a `foldable` method that *should* work. and then there `fold` method is called - which we generally implement using `EvaluatorMapper.fold` which will land you on a different code path - at or, rather, it'll generate the evaluator and then call it, so it might end up on this code path - but it's not worth optimizing this case then. Because it'll only ever call the evaluator one time.

Reply:
I've now pushed a fix that removed all `ConstantAndConstant` methods, classes and generated code, and instead implemented `fold()` correctly. All my tests (unit and integration) are passing, so this does seem like a big improvement.